bin/s2nc.c

```diff
@@ -746,6 +777,16 @@ int main(int argc, char *const *argv)
             GUARD_EXIT(renegotiate(conn, sockfd, reneg_ctx.wait), "Renegotiation failed");
         }
 
+        if (serialize_out) {
+            uint32_t serialize_length = 0;
+            GUARD_EXIT(s2n_connection_serialization_length(conn, &serialize_length), "Failed to get serialized connection length");
+            uint8_t *mem = malloc(serialize_length);
+            GUARD_EXIT_NULL(mem);
+            GUARD_EXIT(s2n_connection_serialize(conn, mem, serialize_length), "Failed to get serialized connection");
+            GUARD_EXIT(write_array_to_file(serialize_out, mem, serialize_length), "Failed to write serialized connection to file");
+            free(mem);
+        }
+
         GUARD_EXIT(wait_for_shutdown(conn, sockfd), "Error closing connection");
```

Comment:
Is wait_for_shutdown just a wait, or does it also send a close_notify? You want to avoid sending a close_notify if you're going to serialize.

Have you tested this with only one connection serializing/de-serializing? Can you document your manual testing in the PR description?

Reply:
That's kind of the issue. I _can't_ test these changes with only one side doing ser/de. s2nc/d are processes, so any TCP connection will be broken if the process exits, as it has to do in order to serialize and then deserialize. It's never going to work with only one side doing ser/de.

I mean, I can move calls around to see what would order would hypothetically be necessary if only one side was doing ser/de, but it's mostly theater since s2nc/d won't actually be able to do that.